app/services/custom_fields/hierarchy/hierarchical_item_service.rb

```diff
@@ -67,13 +82,18 @@ def create_root_item
         Success(item)
       end
 
-      def create_child_item(validation)
-        item = CustomField::Hierarchy::Item
-                 .create(parent: validation[:parent], label: validation[:label], short: validation[:short])
+      def create_child_item(validation:)
+        item = validation[:parent].children.create(label: validation[:label], short: validation[:short])
```

Comment:
🟡 Hmm, not sure, but I do not like this change. What is the benefit? Before the code was very clear, that a new Item is created. Now I must understand, that the item to create is referenced here as `.children`.

🔴 Does this even work, if the created item is not the only child?

Reply:
Because that's how `closure_tree` works. It keeps intermediate representations of the tree for lookup, so we need to go through its code.

From another perspective, it is clearer to the random RoR dev as `children.create` maps to "I'm creating some associated record to the caller". :P

Reply:
Can you confirm, that this does not break, if the parent already has persisted children, when adding a new one?

Reply:
Yes. It works. Why? Because `.children` is an association proxy that don't override anything but ensure that the fields for the association are correctly set - and in case of closure tree, sets up the rest of the fields and optimizations.